lib/src/file_picker_macos.dart

```diff
@@ -107,6 +107,10 @@ class FilePickerMacOS extends FilePicker {
   }
 
   String fileTypeToFileFilter(FileType type, List<String>? allowedExtensions) {
+    if (type != FileType.custom && (allowedExtensions?.isNotEmpty ?? false)) {
+      throw Exception(
```

Comment:
Could you turn this into an `ArgumentError` ? That would make the intent a bit clearer IMO.

And we can tweak the message, to remove the "please" grammar.

```
throw ArgumentError.value(
   allowedExtensions,
   'allowedExtensions',
   'Custom extension filters are only allowed with FileType.custom. '
   'Remove the extension filter or change the FileType to FileType.custom.'
);
```

Reply:
Done - thanks for the review!